bitcoin.go

```diff
@@ -70,43 +69,56 @@ func bitcoinDecodeToBytes(input string, config *CoinConfig) ([]byte, error) {
 		return nil, errors.New("invalid address")
 	}
 	// try base58 first
-	bytes, version, err := base58.CheckDecode(input)
+	data, err := Base58AddressDecodeToBytesPrefix(input, btcKeyHashLenght+len(config.P2SHPPrefix), [][]byte{config.P2PKHPrefix, config.P2SHPPrefix})
 	if err != nil {
 		if len(config.HRP) <= 0 {
 			return nil, err
 		}
 		// try bech32
-		decodedHrp, bytes, err := bech32.Decode(input)
-		if err != nil {
-			return nil, errors.Wrapf(err, "decoding bech32 failed")
+		decodedHrp, data, err2 := bech32.Decode(input)
```

Comment:
it's ok to return only last error
```suggestion
		decodedHrp, data, err := bech32.Decode(input)
```

Reply:
In this 'try both decoding version and see if one succeed', it is useful to have both errors.  Particularly, in the wrong prefix case we get the bech32 error, which is not very useful.
But let's keep it simple.